### PR TITLE
task/JM-6650 remove legacy letter code

### PR DESCRIPTION
**Pull request by akikrahman1**
### JIRA link (if applicable) ###

https://centralgovernmentcgi.atlassian.net/browse/JM-6650

### Change description ###

Updated code to remove references to the legacy letter queue service and use the new printData service.

Updated unit and integration tests

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/ManagePoolControllerITest.java [hunk 1]
```diff
@@ -1075,12 +1082,10 @@ public void test_transferJurorsToNewCourt_oneJuror_firstTransfer_happyPath() thr
         transferJurorPoolValidateNewlyCreatedJurorPool("111111111", sourcePoolNumber,
             "457230702", targetCourtLocation, targetServiceStartDate, COURT_USER);
 
-        long letterCount = certLetterRepository.count();
+        // verify no letters have been queued for bulk print
+        List<BulkPrintData> bulkPrintData = bulkPrintDataRepository.findAll();
+        assertThat(bulkPrintData.size()).isEqualTo(0);
```
Comment (Ben-Edwards-cgi) [inline src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/ManagePoolControllerITest.java:1087]: Why is this checking for 0 when previously it was checking for 1?

File: src/test/java/uk/gov/hmcts/juror/api/moj/service/JurorManagementServiceImplTest.java [hunk 2]
```diff
@@ -510,8 +503,6 @@ public void test_transferPoolMembers_happyPath_onePoolMember_firstTransfer() {
             .saveAndFlush(any());
         verify(jurorHistoryRepository, times(jurorNumbers.size()))
             .save(any());
-        verify(certLetterService, times(jurorNumbers.size()))
-            .enqueueNewLetter(any(), any());
```
Comment (Ben-Edwards-cgi) [inline src/test/java/uk/gov/hmcts/juror/api/moj/service/JurorManagementServiceImplTest.java:514]: Should verify print service is called (Same for all)

**Review by Ben-Edwards-cgi — CHANGES_REQUESTED** [review]

File: src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/ManagePoolControllerITest.java (hunk 1 above)
Comment (akikrahman1) [reply src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/ManagePoolControllerITest.java:1087]: its a court user so no letters should be queued - I added that logic in this MR

File: src/test/java/uk/gov/hmcts/juror/api/moj/service/JurorManagementServiceImplTest.java (hunk 2 above)
Comment (akikrahman1) [reply src/test/java/uk/gov/hmcts/juror/api/moj/service/JurorManagementServiceImplTest.java:514]: we dont need this as its a cert letter, these are not queued after discussion with Chris W - courts can separately print them off

**Review by Ben-Edwards-cgi — APPROVED** [review]

**Pull request merged**